### PR TITLE
GEOMESA-1190 JSON Array Converters

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**
Signed-off-by: Andrew Hulbert <andrew.hulbert@ccri.com>

### CODE REVIEW

File: geomesa-convert/geomesa-convert-json/src/main/scala/org/locationtech/geomesa/convert/json/JsonFunctionFactory.scala [hunk 1]
```diff
@@ -22,3 +23,36 @@ class JsonFunctionFactory extends TransformerFunctionFactory {
...
+class JsonMapListFunctionFactory extends TransformerFunctionFactory with MapListParsing {
+  override def functions = Seq(jsonListParser)
+
+  import scala.collection.JavaConverters._
+
+  def getPrimitive(p: JsonPrimitive) = {
+    if (p.isBoolean) {
+      p.getAsBoolean
+    } else if (p.isNumber) {
+      p.getAsString
```
Comment (elahrvivaz) [inline geomesa-convert/geomesa-convert-json/src/main/scala/org/locationtech/geomesa/convert/json/JsonFunctionFactory.scala]: do you want numbers to be strings? that doesn't match the other json parsing does it?

Comment (jahhulbert-ccri) [inline geomesa-convert/geomesa-convert-json/src/main/scala/org/locationtech/geomesa/convert/json/JsonFunctionFactory.scala]: yeah you only get those three types in JSON unfortunately and yeah then the conversion happens after. I guess we could turn it into a Number class as well...

File: geomesa-convert/geomesa-convert-json/src/test/scala/org/locationtech/geomesa/convert/json/JsonConverterTest.scala [hunk 2]
```diff
@@ -716,5 +716,91 @@ class JsonConverterTest extends Specification {
...
+          |        "geometry": {"type": "Point", "coordinates": [55, 56]},
+          |        "things": [
+          |          {
+          |            "s": "s1",
+          |            "i": 1,
+          |            "d": 1.1,
+          |            "u": "12345678-1234-1234-1234-123456781234"
+          |          },
+                     {
+          |            "s": "s2",
```
Comment (elahrvivaz) [inline geomesa-convert/geomesa-convert-json/src/test/scala/org/locationtech/geomesa/convert/json/JsonConverterTest.scala]: So seems like it's merging fields from a list of objects. Will it also work if you just have a plain list of e.g. ints or something?

Comment (jahhulbert-ccri) [inline geomesa-convert/geomesa-convert-json/src/test/scala/org/locationtech/geomesa/convert/json/JsonConverterTest.scala]: i'll try it out! good idea

Comment (jahhulbert-ccri) [inline geomesa-convert/geomesa-convert-json/src/test/scala/org/locationtech/geomesa/convert/json/JsonConverterTest.scala]: looks like it does :) added a test

**Comment by jahhulbert-ccri** [comment]
@elahrvivaz i named the transform function jsonList...could have made it parseJsonList lemme know what you think

**Comment by elahrvivaz** [comment]
jsonList seems fine. Ready to merge then?

**Comment by jahhulbert-ccri** [comment]
yup should be ready

**Comment by elahrvivaz** [comment]
merging

**Comment by elahrvivaz** [comment]
b63611523c9c953ecef656564c26804d32c03441

**Pull request closed without merging**